server/neptune/sync/exec.go

```diff
@@ -0,0 +1,68 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/logger"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+// RunNewProcessGroupCommand is useful for running separate commands that shouldn't receive termination
+// signals at the same time as the parent process. The passed context should listen for a timeout/cancellation
+// that allows for the parent process to perform a shutdown on its own terms.
+func RunNewProcessGroupCommand(ctx context.Context, cmd *exec.Cmd, cmdName string) error {
```

Comment:
feels weird to force a consumer to pass in a command name just for logging purposes. Would rather leave it out at that point.